### ISSUE TITLE
All backups to B2 Cloud Storage failed

### ISSUE THREAD

**Issue by Ivan4848** [duplicati/duplicati#3330]
## Environment info

- **Duplicati version**:  2.0.3.9_canary_2018-06-30
- **Operating system**: Windows Server 2008 R2
- **Backend**: B2 Cloud Storage

## Description
All backups to B2 Cloud Storage failed. Log after 31 Jul dnt worked. I can make only 1 test.

## Screenshots
![image](https://user-images.githubusercontent.com/33145857/43629090-33101eb8-9737-11e8-9964-c66decc273d6.png)
Backups must make everyday, but after 25 Jul all time failed.

![image](https://user-images.githubusercontent.com/33145857/43629585-a52854c4-9738-11e8-8888-7b1f488aee34.png)

## Debug log now
MainOperation: Test
Verifications: [
    Key: duplicati-20180618T180100Z.dlist.zip.aes
    Value: [],
    Key: duplicati-id82a0bd333eb47d682e1396accf45de5.dindex.zip.aes
    Value: [],
    Key: duplicati-b146c7eec008e41a29f429b2391fd4817.dblock.zip.aes
    Value: []
]
ParsedResult: Success
Version: 2.0.3.9 (2.0.3.9_canary_2018-06-30)
EndTime: 03.08.2018 15:47:17 (1533278837)
BeginTime: 03.08.2018 15:42:50 (1533278570)
Duration: 00:04:27.8933226
Messages: [
    2018-08-03 15:42:50 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Test,
    2018-08-03 15:42:50 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
    2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (416 байт),
    2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180730T200724Z.dlist.zip.aes,
    2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-b537f1860f93341fba8575ca2230a6fe4.dblock.zip.aes,
...
]
Warnings: []
Errors: []
TaskReader:
    ProgressAsync:
        Result: True
        Factory:
            CancellationToken:
                IsCancellationRequested: False
                CanBeCanceled: False
                WaitHandle:
                    Handle: 2540
                    SafeWaitHandle:
                        IsInvalid: False
                        IsClosed: False
            Scheduler: null
            CreationOptions: None
            ContinuationOptions: None
        Id: 11
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False
    TransferProgressAsync:
        Result: True
        Id: 12
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False
BackendStatistics:
    RemoteCalls: 4
    BytesUploaded: 0
    BytesDownloaded: 105519879
    FilesUploaded: 0
    FilesDownloaded: 3
    FilesDeleted: 0
    FoldersCreated: 0
    RetryAttempts: 0
    UnknownFileSize: 0
    UnknownFileCount: 0
    KnownFileCount: 416
    KnownFileSize: 10395408160
    LastBackupDate: 25.07.2018 16:19:49 (1532503189)
    BackupListCount: 144
    TotalQuotaSpace: 0
    FreeQuotaSpace: 0
    AssignedQuotaSpace: -1
    ReportedQuotaError: False
    ReportedQuotaWarning: False
    ParsedResult: Success
    Version: 2.0.3.9 (2.0.3.9_canary_2018-06-30)
    Messages: [
        2018-08-03 15:42:50 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Test,
        2018-08-03 15:42:50 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (416 байт),
        2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180730T200724Z.dlist.zip.aes,
        2018-08-03 15:46:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-b537f1860f93341fba8575ca2230a6fe4.dblock.zip.aes,
...
    ]
    Warnings: []
    Errors: []
    TaskReader: null

## Log 31 Jul
MainOperation: Repair
RecreateDatabaseResults: null
ParsedResult: Success
EndTime: 31.07.2018 4:44:13 (1532979853)
BeginTime: 31.07.2018 4:40:57 (1532979657)
Duration: 00:03:16.4022335
Messages: [
    2018-07-31 04:40:57 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Repair,
    2018-07-31 04:40:58 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
    2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (416 байт),
    2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180727T180100Z.dlist.zip.aes,
    2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-beb800e80493a4a2f94b3438c3b9678af.dblock.zip.aes,
...
]
Warnings: []
Errors: []
TaskReader:
    ProgressAsync:
        Result: True
        Factory:
            CancellationToken:
                IsCancellationRequested: False
                CanBeCanceled: False
                WaitHandle:
                    Handle: 3708
                    SafeWaitHandle:
                        IsInvalid: False
                        IsClosed: False
            Scheduler: null
            CreationOptions: None
            ContinuationOptions: None
        Id: 13
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False
    TransferProgressAsync:
        Result: True
        Id: 14
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False
BackendStatistics:
    RemoteCalls: 1
    BytesUploaded: 0
    BytesDownloaded: 0
    FilesUploaded: 0
    FilesDownloaded: 0
    FilesDeleted: 0
    FoldersCreated: 0
    RetryAttempts: 0
    UnknownFileSize: 0
    UnknownFileCount: 0
    KnownFileCount: 416
    KnownFileSize: 10395408160
    LastBackupDate: 25.07.2018 16:19:49 (1532503189)
    BackupListCount: 144
    TotalQuotaSpace: 0
    FreeQuotaSpace: 0
    AssignedQuotaSpace: -1
    ReportedQuotaError: False
    ReportedQuotaWarning: False
    ParsedResult: Success
    Messages: [
        2018-07-31 04:40:57 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Repair,
        2018-07-31 04:40:58 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (416 байт),
        2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180727T180100Z.dlist.zip.aes,
        2018-07-31 04:44:09 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-beb800e80493a4a2f94b3438c3b9678af.dblock.zip.aes,
...
    ]
    Warnings: []
    Errors: []
    TaskReader: null

## Log 27 Jul
Jul 27, 2018 9:00 AM: Result
DeletedFiles: 3
DeletedFolders: 0
ModifiedFiles: 12
ExaminedFiles: 4332
OpenedFiles: 29
AddedFiles: 17
SizeOfModifiedFiles: 0
SizeOfAddedFiles: 0
SizeOfExaminedFiles: 36269481276
SizeOfOpenedFiles: 16426443275
NotProcessedFiles: 0
AddedFolders: 6
TooLargeFiles: 0
FilesWithError: 0
ModifiedFolders: 0
ModifiedSymlinks: 0
AddedSymlinks: 0
DeletedSymlinks: 0
PartialBackup: False
Dryrun: False
MainOperation: Backup
CompactResults:
    DeletedFileCount: 0
    DownloadedFileCount: 0
    UploadedFileCount: 0
    DeletedFileSize: 0
    DownloadedFileSize: 0
    UploadedFileSize: 0
    Dryrun: False
    MainOperation: Compact
    ParsedResult: Warning
    EndTime: 27.07.2018 8:53:51 (1532649231)
    BeginTime: 27.07.2018 8:53:17 (1532649197)
    Duration: 00:00:33.8299349
    Messages: [
        2018-07-25 16:19:49 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Backup,
        2018-07-25 16:20:16 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (404 байт),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180724T180100Z.dlist.zip.aes,
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-SchedulingMissingFileForDelete]: scheduling missing file for deletion, currently listed as Uploading: duplicati-b6594da301be145bfaf3c743e0dc11934.dblock.zip.aes,
...
    ]
    Warnings: [
        2018-07-25 16:23:23 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.UploadSyntheticFilelist-MissingTemporaryFilelist]: Expected there to be a temporary fileset for synthetic filelist (230, duplicati-b97dc0f0d416d4d2aa8ad19292c9c00b3.dblock.zip.aes), but none was found?,
        2018-07-27 03:12:31 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.FileBlockProcessor.FileEntry-PathProcessingFailed]: Failed to process path: D:\DB\ВТМ_zao\1Cv8tmp.1CL
    ]
    Errors: []
    TaskReader: null
    BackendStatistics:
        RemoteCalls: 26
        BytesUploaded: 677723488
        BytesDownloaded: 105323911
        FilesUploaded: 16
        FilesDownloaded: 3
        FilesDeleted: 4
        FoldersCreated: 0
        RetryAttempts: 1
        UnknownFileSize: 0
        UnknownFileCount: 0
        KnownFileCount: 416
        KnownFileSize: 10395408160
        LastBackupDate: 25.07.2018 16:19:49 (1532503189)
        BackupListCount: 144
        TotalQuotaSpace: 0
        FreeQuotaSpace: 0
        AssignedQuotaSpace: -1
        ReportedQuotaError: False
        ReportedQuotaWarning: False
        ParsedResult: Warning
        Messages: [
            2018-07-25 16:19:49 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Backup,
            2018-07-25 16:20:16 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
            2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (404 байт),
            2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180724T180100Z.dlist.zip.aes,
            2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-SchedulingMissingFileForDelete]: scheduling missing file for deletion, currently listed as Uploading: duplicati-b6594da301be145bfaf3c743e0dc11934.dblock.zip.aes,
...
        ]
        Warnings: [
            2018-07-25 16:23:23 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.UploadSyntheticFilelist-MissingTemporaryFilelist]: Expected there to be a temporary fileset for synthetic filelist (230, duplicati-b97dc0f0d416d4d2aa8ad19292c9c00b3.dblock.zip.aes), but none was found?,
            2018-07-27 03:12:31 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.FileBlockProcessor.FileEntry-PathProcessingFailed]: Failed to process path: D:\DB\ВТМ_zao\1Cv8tmp.1CL
        ]
        Errors: []
        TaskReader: null
DeleteResults:
    DeletedSets: [
        Item1: 144
        Item2: 27.01.2018 4:01:00 (1516993260),
        Item1: 145
        Item2: 26.01.2018 4:01:00 (1516906860),
        Item1: 146
        Item2: 25.01.2018 4:01:00 (1516820460),
        Item1: 147
        Item2: 24.01.2018 4:01:00 (1516734060)
    ]
    Dryrun: False
    MainOperation: Delete
    ParsedResult: Warning
    EndTime: 27.07.2018 8:53:51 (1532649231)
    BeginTime: 27.07.2018 8:52:50 (1532649170)
    Duration: 00:01:00.1474403
    Messages: [
        2018-07-25 16:19:49 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Backup,
        2018-07-25 16:20:16 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (404 байт),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180724T180100Z.dlist.zip.aes,
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-SchedulingMissingFileForDelete]: scheduling missing file for deletion, currently listed as Uploading: duplicati-b6594da301be145bfaf3c743e0dc11934.dblock.zip.aes,
...
    ]
    Warnings: [
        2018-07-25 16:23:23 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.UploadSyntheticFilelist-MissingTemporaryFilelist]: Expected there to be a temporary fileset for synthetic filelist (230, duplicati-b97dc0f0d416d4d2aa8ad19292c9c00b3.dblock.zip.aes), but none was found?,
        2018-07-27 03:12:31 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.FileBlockProcessor.FileEntry-PathProcessingFailed]: Failed to process path: D:\DB\ВТМ_zao\1Cv8tmp.1CL
    ]
    Errors: []
    TaskReader: null
RepairResults: null
TestResults:
    MainOperation: Test
    Verifications: [
        Key: duplicati-20180725T071949Z.dlist.zip.aes
        Value: [],
        Key: duplicati-ib824161bdb234a2abc42580c6e169e29.dindex.zip.aes
        Value: [],
        Key: duplicati-b8eb74ead2e804bc99fe4cda88841c6ff.dblock.zip.aes
        Value: []
    ]
    ParsedResult: Warning
    EndTime: 27.07.2018 9:00:53 (1532649653)
    BeginTime: 27.07.2018 8:56:52 (1532649412)
    Duration: 00:04:01.2657996
    Messages: [
        2018-07-25 16:19:49 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Backup,
        2018-07-25 16:20:16 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (404 байт),
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180724T180100Z.dlist.zip.aes,
        2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-SchedulingMissingFileForDelete]: scheduling missing file for deletion, currently listed as Uploading: duplicati-b6594da301be145bfaf3c743e0dc11934.dblock.zip.aes,
...
    ]
    Warnings: [
        2018-07-25 16:23:23 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.UploadSyntheticFilelist-MissingTemporaryFilelist]: Expected there to be a temporary fileset for synthetic filelist (230, duplicati-b97dc0f0d416d4d2aa8ad19292c9c00b3.dblock.zip.aes), but none was found?,
        2018-07-27 03:12:31 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.FileBlockProcessor.FileEntry-PathProcessingFailed]: Failed to process path: D:\DB\ВТМ_zao\1Cv8tmp.1CL
    ]
    Errors: []
    TaskReader: null
ParsedResult: Warning
EndTime: 27.07.2018 9:00:53 (1532649653)
BeginTime: 25.07.2018 16:19:48 (1532503188)
Duration: 1.16:41:05.2196593
Messages: [
    2018-07-25 16:19:49 +09 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: Началась операция Backup,
    2018-07-25 16:20:16 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Started:  (),
    2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.BasicResults-BackendEvent]: Backend event: List - Completed:  (404 байт),
    2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-RemoteUnwantedMissingFile]: removing file listed as Temporary: duplicati-20180724T180100Z.dlist.zip.aes,
    2018-07-25 16:23:21 +09 - [Information-Duplicati.Library.Main.Operation.FilelistProcessor-SchedulingMissingFileForDelete]: scheduling missing file for deletion, currently listed as Uploading: duplicati-b6594da301be145bfaf3c743e0dc11934.dblock.zip.aes,
...
]
Warnings: [
    2018-07-25 16:23:23 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.UploadSyntheticFilelist-MissingTemporaryFilelist]: Expected there to be a temporary fileset for synthetic filelist (230, duplicati-b97dc0f0d416d4d2aa8ad19292c9c00b3.dblock.zip.aes), but none was found?,
    2018-07-27 03:12:31 +09 - [Warning-Duplicati.Library.Main.Operation.Backup.FileBlockProcessor.FileEntry-PathProcessingFailed]: Failed to process path: D:\DB\ВТМ_zao\1Cv8tmp.1CL
]
Errors: []
TaskReader:
    ProgressAsync:
        Result: True
        Factory:
            CancellationToken:
                IsCancellationRequested: False
                CanBeCanceled: False
                WaitHandle:
                    Handle: 3708
                    SafeWaitHandle:
                        IsInvalid: False
                        IsClosed: False
            Scheduler: null
            CreationOptions: None
            ContinuationOptions: None
        Id: 1
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False
    TransferProgressAsync:
        Result: True
        Id: 2
        Exception: null
        Status: RanToCompletion
        IsCanceled: False
        IsCompleted: True
        CreationOptions: None
        AsyncState: null
        IsFaulted: False

## Full bugreport
_bugreport.zip_
https://drive.google.com/drive/folders/1MVydObjD1u2yzY4On7o1uwie2Auz_kwV?usp=sharing

pls help

**Comment by warwickmm** [comment]
It's a bit strange for the backup to suddenly stop working.

* Is it possible that you've run out of space in your B2 storage?
* Are you able to create a new backup with some other destination (e.g., local) and does that run successfully?

**Comment by Ivan4848** [comment]
> Is it possible that you've run out of space in your B2 storage?

No

> Are you able to create a new backup with some other destination (e.g., local) and does that run successfully?

Also does not work for local.

**Comment by Ivan4848** [comment]
Problem solved
1) Deleted duplicates in the system
2) Deleted all files for the mask: *duplicati *
3) Install las version Duplicati
4) Re-create backup record
all work

The problem was in the remaining files after the deletion Duplicati in the system. 
I installed a new version Duplicati on the old Duplicati files (mask *duplicati*) and this caused a crash backuping.
When I deleted all files *duplicati* after delete Duplicati in system and install last version - all work good.
It's inconvenient to re-create a backup with each new version of Duplicati.

The problem was on two servers.
And there I solved the problem the same way.
